test/plots/stackedBarPlotTests.ts

```diff
@@ -1,607 +1,411 @@
+            let yData = dataset.data().map((d, i) => yAccessor(d, i, dataset));
+            if (datasetIndex === 0) {
+              stackedYDataArray[datasetIndex] = yData;
+              return;
+            }
+            stackedYDataArray[datasetIndex] = dataset.data().map((d, i) => {
+              return yData[i] + stackedYDataArray[datasetIndex - 1][i];
+            });
+          });
+          return stackedYDataArray;
+        };
+
+        let stackedYs = calculateStackedYs((d) => d.y);
+        bars.each(function(d, i) {
+          let bar = d3.select(this);
+          let datasetIndex = Math.floor(i / dataLength);
+          let datumIndex = i % dataLength;
+          assert.closeTo(TestMethods.numAttr(bar, "y"), yScale.scale(stackedYs[datasetIndex][datumIndex]),
+            window.Pixel_CloseTo_Requirement, "y attribute offset set correctly");
+        });
+
+        stackedBarPlot.destroy();
+        svg.remove();
       });
 
-      let dataset2Rects = plot.content().selectAll("." + dataset2.metadata().id);
-      dataset2Rects.each(function (datum, index) {
-        let rect = d3.select(this);
-        let expectedY = yScale.scale(data2[index].y + data1[index].y);
-        let actualY = TestMethods.numAttr(rect, "y");
-        assert.closeTo(actualY, expectedY, 0.1, "y attribute set correctly (dataset 2, datum " + index + ")");
-        let expectedHeight = yScale.scale(data1[index].y) - expectedY;
-        let actualHeight = TestMethods.numAttr(rect, "height");
-        assert.closeTo(actualHeight, expectedHeight, 0.1, "height attribute set correctly (dataset 2, datum " + index + ")");
+      // HACKHACK #2795: correct off-bar label logic to be implemented
+      it("doesn't show any off-bar labels", () => {
+        stackedBarPlot.labelsEnabled(true);
+        yScale.domain([0, 30]);
+        let offBarLabels = stackedBarPlot.content().selectAll(".off-bar-label");
+        assert.operator(offBarLabels.size(), ">", 0, "some off-bar labels are drawn");
+        let offBarLabelVisibilities: string[] = [];
+        offBarLabels.each(function() {
+          offBarLabelVisibilities.push(d3.select(this).style("visibility"));
+        });
+        assert.isTrue(offBarLabelVisibilities.every((visibility) => visibility === "hidden"), "all off-bar labels are hidden");
+        stackedBarPlot.destroy();
+        svg.remove();
       });
 
-      svg.remove();
+      it("considers lying within a bar's y-range to mean it is closest", () => {
+        let bars = stackedBarPlot.content().selectAll("rect");
+
+        let d0 = stackedBarPlot.datasets()[0].data()[0];
+        let d0Px = {
+          x: xScale.scale(d0.x),
+          y: yScale.scale(d0.y)
+        };
+
+        let d1 = stackedBarPlot.datasets()[1].data()[0];
+        let d1Px = {
+          x: xScale.scale(d1.x),
+          y: yScale.scale(d0.y + d1.y)
+        };
+
+        let expected: Plottable.Plots.PlotEntity = {
```

Comment:
is it necessary to create a whole entity? is checking datum/index of the selected entity not sufficient?